package.json

```diff
@@ -71,12 +72,15 @@
     "release": "git pull && git push origin master --tags && npm publish",
     "release-legacy": "git pull origin 5.x && git push origin 5.x --tags && npm publish --tag legacy",
     "mongo": "node ./tools/repl.js",
-    "test": "mocha --exit ./test/*.test.js ./test/typescript/main.test.js",
+    "test": "npm run test-code && npm run test-ts-integration && npm run test-ts-types",
```

Comment:
As written, I don't think this will support passing command line args to mocha, like `npm test -- -g "regexp here"`.

Ideally, I'd rather the TS tests run separately

Reply:
For this case you would do npm run test-code -- -g "regexp here"

But if you prefer, I can remove the current test and rename test-code to test.